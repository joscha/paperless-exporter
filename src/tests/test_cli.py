import sys
import subprocess
from pathlib import Path

from . import assert_tree_snapshot


def test_cli_generates_expected_tree(tmp_path, snapshot):
    source = Path("fixtures/library.paperless")
    target = tmp_path / "out"
    result = subprocess.run(
        [sys.executable, "-m", "src.cli", str(source), str(target)],
        capture_output=True,
        text=True,
    )
    # Optionally print result.stdout/result.stderr for debugging
    assert snapshot(name="stdout") == result.stdout
    assert snapshot(name="stderr") == result.stderr
    assert_tree_snapshot(target, snapshot(name="generated_tree"))
