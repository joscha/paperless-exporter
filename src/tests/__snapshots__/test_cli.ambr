# serializer version: 1
# name: test_cli_generates_expected_tree[generated_tree]
  '''
  out/
  ├── _attachments/
  │   ├── 2025-04-18_01_sample-invoice.document.pdf
  │   ├── 2025-04-18_02_wordpress-pdf-invoice-plugin-sample.document.pdf
  │   ├── 2025-04-24_03_sample-invoice.document.pdf
  │   ├── 2025-04-24_03_sample-invoice.original.pdf
  │   ├── 2025-04-25_04_missing.thumbnail.png
  │   ├── 2025-05-01_05_pdf-test.document.pdf
  │   ├── 2025-05-01_06_drylab.document.pdf
  │   ├── 2025-05-01_07_invoicesample.document.pdf
  │   ├── 2025-05-01_08_PDF_TestPage.document.pdf
  │   ├── 2025-05-01_09_Get_Started_With_Smallpdf.document.pdf
  │   ├── 2025-05-01_10_Free_Test_Data_100KB_PDF.document.pdf
  │   ├── 2025-05-01_11_test_pdf_acrobat.document.pdf
  │   ├── 2025-05-01_12_sample_ _with_ special characters _o_ _.document.pdf
  │   └── 2025-05-01_13_file-sample_150kB_mit_Ümläutön.document.pdf
  ├── collections/
  │   ├── Test Folder/
  │   │   └── Another collection.md
  │   ├── Library.md
  │   ├── My collection.md
  │   └── New Test PDFs.md
  ├── drylab (6).md
  ├── file-sample_150kB_mit_Ümläutön (13).md
  ├── Free_Test_Data_100KB_PDF (10).md
  ├── Get_Started_With_Smallpdf (9).md
  ├── invoicesample (7).md
  ├── missing (4).md
  ├── pdf-test (5).md
  ├── PDF_TestPage (8).md
  ├── sample-invoice (1).md
  ├── sample-invoice (3).md
  ├── sample_ _with_ special characters _o_ _ (12).md
  ├── test_pdf_acrobat (11).md
  └── wordpress-pdf-invoice-plugin-sample (2).md
  
  '''
# ---
# name: test_cli_generates_expected_tree[stderr]
  '''
  No documents exist for receipt 'missing' (Paperless ID: 4; Document path: fixtures/library.paperless/Documents/2025/04/25/07D1AB33-E307-4274-9DBE-C - 25-4-2025.pdf).
  
  '''
# ---
# name: test_cli_generates_expected_tree[stdout]
  '''
  Export completed successfully.
  
  '''
# ---
